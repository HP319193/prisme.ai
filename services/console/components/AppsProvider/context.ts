import { createContext, useContext } from 'react';

export interface AppsContext {
  apps: Map<string, Prismeai.App>;
  appInstances: Map<string, Prismeai.DetailedAppInstance[]>;
  getApps: ({
    text,
    page,
    limit,
    workspaceId,
  }: {
    text?: PrismeaiAPI.SearchApps.QueryParameters['text'];
    page?: PrismeaiAPI.SearchApps.QueryParameters['page'];
    limit?: PrismeaiAPI.SearchApps.QueryParameters['limit'];
    workspaceId?: PrismeaiAPI.SearchApps.QueryParameters['workspaceId'];
  }) => Promise<Prismeai.App[] | null>;
  getAppInstances: (
    workspaceId: PrismeaiAPI.ListAppInstances.PathParameters['workspaceId']
  ) => Promise<Prismeai.DetailedAppInstance[]>;
}

export const appsContext = createContext<AppsContext>({
  apps: new Map(),
  appInstances: new Map(),
  getApps: async () => ({} as any),
  getAppInstances: async () => ({} as any),
});

export const useApps = () => useContext(appsContext);

export default appsContext;
