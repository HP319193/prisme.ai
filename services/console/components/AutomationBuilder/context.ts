import { createContext, useContext } from 'react';
import { Schema } from '../SchemaForm/types';

export interface AutomationBuilderContext {
  addInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  removeInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editInstruction: (parent: Prismeai.InstructionList, index: number) => void;
  editCondition: (
    parent: { conditions: Prismeai.Conditions },
    key: string
  ) => void;
  editTrigger: () => void;
  getApp: (
    instruction: string
  ) => {
    name: string;
    icon: string;
  };
  instructionsSchemas: [string, Record<string, Schema>, { icon: string }][];
  getSchema: (name: string) => Schema;
}
export const automationBuilderContext = createContext<AutomationBuilderContext>(
  {
    addInstruction() {},
    removeInstruction() {},
    editInstruction() {},
    editCondition() {},
    editTrigger() {},
    getApp() {
      return {
        name: '',
        icon: '',
      };
    },
    instructionsSchemas: [],
    getSchema() {
      return {};
    },
  }
);

export const useAutomationBuilder = () => useContext(automationBuilderContext);

export default automationBuilderContext;
