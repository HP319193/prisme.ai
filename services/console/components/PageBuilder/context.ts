import { createContext } from 'react';
import { Schema } from '@prisme.ai/design-system';
import { BlockInCatalog } from './useBlocks';
import { useContext } from '../../utils/useContext';

export type Block = NonNullable<Prismeai.Page['blocks']>[number];
export type BlocksWithKeys = Map<string, Block>;

export interface PageBuilderContext {
  value: BlocksWithKeys;
  addBlock: (position: number, blockName?: string) => void;
  setEditBlock: (blockId: string) => void;
  removeBlock: (key: string) => void;
  setBlockConfig: (key: string, config: any) => void;
  setBlockSchema: (blockId: string, schema: Schema | null) => void;
  catalog: BlockInCatalog[];
  blocksSchemas: Map<string, Schema | null>;
}

export const context = createContext<PageBuilderContext | undefined>(undefined);

export const usePageBuilder = () => useContext<PageBuilderContext>(context);
