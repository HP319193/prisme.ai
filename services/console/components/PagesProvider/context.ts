import { createContext, Dispatch, SetStateAction, useContext } from 'react';

type WorkspaceId = string;

export interface PagesContext {
  pages: Map<WorkspaceId, Set<Prismeai.Page>>;
  fetchPages: (workspaceId: NonNullable<Prismeai.Workspace['id']>) => void;
  setPages: Dispatch<SetStateAction<Map<string, Set<Prismeai.Page>>>>;
  createPage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  savePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    p: Prismeai.Page
  ) => Promise<Prismeai.Page>;
  deletePage: (
    workspaceId: NonNullable<Prismeai.Workspace['id']>,
    pid: NonNullable<Prismeai.Page['id']>
  ) => Promise<Pick<Prismeai.Page, 'id'>>;
}

export const pagesContext = createContext<PagesContext>({
  pages: new Map(),
  setPages() {},
  fetchPages() {},
  async createPage() {
    return {} as Prismeai.Page;
  },
  async savePage() {
    return {} as Prismeai.Page;
  },
  async deletePage() {
    return {} as Prismeai.Page;
  },
});

export const usePages = () => useContext(pagesContext);

export default usePages;
