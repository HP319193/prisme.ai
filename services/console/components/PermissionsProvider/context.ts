import { createContext, useContext } from 'react';
import UserPermissions = Prismeai.UserPermissions;
import SubjectType = PrismeaiAPI.GetPermissions.Parameters.SubjectType;

export interface PermissionsContext {
  usersPermissions: Map<string, UserPermissions[]>;
  addUserPermissions: (
    subjectType: SubjectType,
    subjectId: string,
    permissions: UserPermissions
  ) => Promise<any>;
  // unshareWorkspaceToUser: (
  //   userEmail: string,
  //   workspaceId: string
  // ) => Promise<any>;
  getUsersPermissions: (
    subjectType: SubjectType,
    subjectId: string
  ) => Promise<UserPermissions[]>;
}

export const workspacesContext = createContext<PermissionsContext>({
  usersPermissions: new Map(),
  // unshareWorkspaceToUser: async () => ({} as any),
  addUserPermissions: async () => ({} as UserPermissions),
  getUsersPermissions: async () => [] as UserPermissions[],
});

export const usePermissions = () => useContext(workspacesContext);

export default workspacesContext;
