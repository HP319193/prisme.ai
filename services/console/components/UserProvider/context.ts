import { createContext, useContext } from 'react';
import { ApiError } from '@prisme.ai/sdk';

export enum OperationSuccess {
  emailSent = 'emailSent',
  passwordReset = 'passwordReset',
}

export interface ApiSuccess {
  type: OperationSuccess;
}
export interface UserContext<T = Prismeai.User | null> {
  user: T;
  loading: boolean;
  error?: ApiError;
  success?: ApiSuccess;
  signin: (email: string, password: string) => Promise<Prismeai.User | null>;
  signup: (
    email: string,
    password: string,
    firstName: string,
    lastName: string
  ) => Promise<Prismeai.User | null>;
  signout: (onServer?: boolean) => void;
  sendPasswordResetMail: (email: string, language: string) => Promise<any>;
  passwordReset: (token: string, password: string) => Promise<any>;
}

export const userContext = createContext<UserContext>({
  user: null,
  loading: false,
  signin: async () => null,
  signup: async () => null,
  signout() {},
  sendPasswordResetMail: async () => null,
  passwordReset: async () => null,
});

export function useUser(throwIfNotExist?: boolean): UserContext;
export function useUser(throwIfNotExist?: true): UserContext<Prismeai.User>;
export function useUser(throwIfNotExist?: boolean) {
  const context = useContext(userContext);

  if (throwIfNotExist && !context.user) {
    throw new Error();
  }

  return context;
}

export default userContext;
