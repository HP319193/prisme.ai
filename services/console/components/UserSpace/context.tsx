import { createContext } from 'react';
import { useContext } from '../../utils/useContext';

export interface UserSpaceConfig {
  /**
   * Main logo alternative
   */
  mainLogo?: {
    /**
     * Logo URL
     */
    url: string;
    /**
     * Attributes to apply on <img /> tag like alt or title
     */
    attrs?: object;
  };
  /**
   * Main URL on main logo link
   * @example: /product/ai-knowledge-chat/assistant?id=6474c0db33959b6283770367
   */
  mainUrl?: string;
  /**
   * The studio can act as a kiosk to keep the user in a URL template
   * @example: /product/ai-knowledge-chat/
   */
  kiosk?: string;
  /**
   * Define if Whats new popover is visible.
   * @default true
   */
  displayWhatsNew?: boolean;
  /**
   * Define if Help popover is visible.
   * @default true
   */
  displayHelp?: boolean;
  /**
   * Define if products popover is visible.
   * @default true
   */
  displayProducts?: boolean;

  /**
   * style object that will be injected inside various sections, allowing to cutomize theme colors ...
   */
  style?: {
    root?: object;
  };

  /**
   * Disable Builder
   */
  disableBuilder?: true;
}

interface UserSpaceProviderContext extends UserSpaceConfig {}
export const userSpaceContext = createContext<
  UserSpaceProviderContext | undefined
>(undefined);
export const useUserSpace = () =>
  useContext<UserSpaceProviderContext>(userSpaceContext);
export default userSpaceContext;
