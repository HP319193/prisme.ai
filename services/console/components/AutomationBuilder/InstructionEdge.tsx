import React, { FC } from 'react';
import {
  EdgeProps,
  getEdgeCenter,
} from 'react-flow-renderer';
import { useAutomationBuilder } from './context';
import Edge from './Edge';

const foreignObjectSize = 40;

export const InstructionEdge: FC<EdgeProps> = (props) => {
  const { addInstruction } = useAutomationBuilder();
  const {
    sourceX,
    sourceY,
    targetX,
    targetY,
    data
  } = props

  const [edgeCenterX, edgeCenterY] = getEdgeCenter({
    sourceX,
    sourceY,
    targetX,
    targetY,
  });

  return (
    <Edge {...props}>

      <foreignObject
        width={foreignObjectSize}
        height={foreignObjectSize}
        x={edgeCenterX - foreignObjectSize / 2}
        y={edgeCenterY - foreignObjectSize / 2}
        className="edgebutton-foreignobject"
        requiredExtensions="http://www.w3.org/1999/xhtml"
      >
        <div
          className="flex justify-content-center align-items-center"
          style={{ height: `${foreignObjectSize}px` }}>
          <button
            className="pi pi-plus border-none bg-primary p-1 border-round z-1 text-xs cursor-pointer"
            onClick={() => addInstruction(data.parent, data.index)} />
        </div>
      </foreignObject>
    </Edge>
  );
}

export default InstructionEdge
