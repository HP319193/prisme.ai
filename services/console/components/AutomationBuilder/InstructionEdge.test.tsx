import InstructionEdge from "./InstructionEdge";
import renderer from "react-test-renderer";
import { Position } from "react-flow-renderer";
import { useAutomationBuilder } from "./context";

jest.mock('./context', () => {
  const mock = {}
  return {
    useAutomationBuilder: () => mock
  }
})
it("should render", () => {
  const root = renderer.create(<InstructionEdge id="a" data={{}} source="b" target="c" sourcePosition={Position.Top} sourceX={0} sourceY={0} targetPosition={Position.Bottom} targetX={10} targetY={10} />);
  expect(root.toJSON()).toMatchSnapshot();
});

it("should add instruction", () => {
  const parent = {};
  useAutomationBuilder().addInstruction = jest.fn();
  const root = renderer.create(<InstructionEdge id="a" data={{
    parent, index: 1
  }} source="b" target="c" sourcePosition={Position.Top} sourceX={0} sourceY={0} targetPosition={Position.Bottom} targetX={10} targetY={10} />);
  root.root.findByType('button').props.onClick();
  expect(useAutomationBuilder().addInstruction).toHaveBeenCalledWith(parent, 1);
});
