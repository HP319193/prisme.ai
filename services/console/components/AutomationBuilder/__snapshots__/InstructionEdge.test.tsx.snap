// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
Array [
  <path
    className="react-flow__edge-path"
    d="M 0,0L 0,0Q 0,5 5,5L 5,5Q 10,5 10,10L 10,10"
    id="a"
    markerEnd="none"
    style={Object {}}
  />,
  <foreignObject
    className="edgebutton-foreignobject"
    height={40}
    requiredExtensions="http://www.w3.org/1999/xhtml"
    width={40}
    x={-15}
    y={-15}
  >
    <div
      className="flex justify-content-center align-items-center"
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <button
        className="pi pi-plus border-none bg-primary p-1 border-round z-1 text-xs cursor-pointer"
        onClick={[Function]}
      />
    </div>
  </foreignObject>,
]
`;
